### PR TITLE
Mikemhenry patch 1

### PR DESCRIPTION
**Pull request by mikemhenry**
Okay so this was a bit more of a quagmire than I thought it would be. I will pin the full list later but wanted to start recording the errors I'm getting, which are surprising since I did pin `freud`. Let me know if I should change the `freud` version to fix these errors. If its not an easy version pin, then I think we should merge in this PR (haha) once I do a full pin and then figure out how to fix these issues:

1. All of the "Seminars and Demonstrations" links 404
2. `module_intros/Box-Box.ipynb`:
cell 3:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-d50910b63a40> in <module>
     10 
     11 # Automatic matrix detection using from_box
---> 12 box = freud.box.Box.from_box([[1, 1, 0], [0, 1, 0.5], [0, 0, 0.5]])
     13 print("The box matrix was automatically detected: \n\t{}\n".format(box))
     14 

box.pxi in freud._freud.Box.from_box()

box.pxi in freud._freud.Box.__cinit__()

TypeError: must be real number, not list
```
cell 16 needs a `import numpy as np`

3. `module_intros/Box-ParticleBuffer.ipynb`:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-e01b4e018a63> in <module>
----> 1 pbuff = freud.box.ParticleBuffer(box)
      2 pbuff.compute(points, 6, images=True)
      3 print(pbuff.buffer_particles[:10], '...')

AttributeError: module 'freud.box' has no attribute 'ParticleBuffer'
```

4. `module_intros/Density-FloatCF.ipynb`:
cell 5
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-b72bd4f4879d> in <module>
      1 fcf = freud.density.FloatCF(rmax=0.5*box.Lx/num_grains, dr=dx)
----> 2 fcf.reset()

AttributeError: 'freud._freud.FloatCF' object has no attribute 'reset'
```
5. `module_intros/Density-LocalDensity.ipynb`:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-20e0b2e031ff> in <module>
      1 density = freud.density.LocalDensity(ref_radius, area, radius)
----> 2 num_neighbors = density.compute(box, ref_points, points).num_neighbors
      3 densities = density.compute(box, ref_points, points).density

density.pxi in freud._freud.LocalDensity.compute()

locality.pxi in freud._freud.make_default_nlist()

locality.pxi in freud._freud.LinkCell.__cinit__()

RuntimeError: Cannot generate a cell list where cell_width is larger than half the box.
```

Ugh, my mybinder kernel died, so I didn't get them all. I think we should add running these notebooks as part of CI, maybe add it to the `freud` CI to run them all, then we can catch if anything breaks the examples.

### CODE REVIEW

**Comment by mikemhenry** [comment]
Looks like that fixed some of those errors, it is too bad mybinder doesn't like the more explicit definition for packages, but I guess they only need to build these images once per repo, so maybe it doesn't matter... 

tl;dr the binder works now, use this link to test:
https://mybinder.org/v2/gh/mikemhenry/freud-examples/mikemhenry-patch-1?filepath=index.ipynb

**Comment by mikemhenry** [comment]
So I kinda went down a rabbit hole but I think `pytest -v --nbval --nbval-lax` works to test notebooks, `pip install pytest nbval` will install it, and the `--nbval-lax` option will check if the cells run without error. Without that option, it checks to see if the output matches what the notebook had.

**Comment by vyasr** [comment]
@mikemhenry any more luck? I'm open to running notebooks as part of CI, I know mbuild has been working to do that so if you've worked out the kinks I'd be happy to piggyback on that work. 

We do need to significantly rewrite our examples as part of our 2.0 rewrite, so as we start writing those we could start testing them as well.

**Comment by mikemhenry** [comment]
It works now if you check that link, I will do another PR on the main repo to add the CI

**Comment by vyasr** [comment]
Sweet thanks!

**Comment by vyasr** [comment]
@mikemhenry wait also how old are these things? I just noticed that the error message shows `box.pxi`, but freud switched from using pxi to pyx files over a year ago now...

**Comment by mikemhenry** [comment]
OP error messages are out of date, I was doing something silly. It will be easier to catch the errors when I add the next PR that tests the notebooks using `nbval`

**Comment by bdice** [comment]
@mikemhenry Somehow I totally missed this PR and discussion. I am rewriting examples now, and that may change our dependencies substantially. I'll be sure to test Binder after we release v2.0 -- the notebooks will soon depend on that (temporary unreleased) version.

**Comment by mikemhenry** [comment]
Hey no worries! I will close this, ping me if you want help setting up the mybinder or automating notebook testing!
